mytb/ipython.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import absolute_import
+from __future__ import print_function
+
+# #############################################################################
+# Copyright : (C) 2019 by Teledomic.eu All rights reserved
+#
+# Name:         mytb.ipython
+#
+# Description:  helper to create an interactive python shell
+#
+# Ipython will be used if installed
+#
+# #############################################################################
+
+
+try:
+    import IPython
+    HAVE_IPYTHON = True
+except Exception:
+    HAVE_IPYTHON = False
+    raise
+
+
+def simple_shell(ctx):
+    """ basic interactive shell if ipython is not found
+    """
+    try:
+        import readline  # noqa: F401
+    except Exception:
+        print("readline not found")
+    import code
+
+    shell = code.InteractiveConsole(ctx)
+    shell.push("from __future__ import print_function")
+    shell.push("from __future__ import absolute_import")
+    shell.interact()
+
+
+def start_shell(ctx=None):
+    """ starts an interactive python shell
+    """
+    ctx = ctx if ctx is not None else {}
+    if HAVE_IPYTHON:
+        IPython.start_ipython(argv=[], user_ns=ctx)
```

Comment:
Yes, this is intended to start ipython embedded within an application in this sense I have to force argv explicly to [] so that  sys,argv of the embedding program is ignored as it might
have either options, which ipython doesn't know or options which have a different meaning for the main program

Reply:
ok, I get it. Will it have any impact on a software using a Django shell for example?